### PR TITLE
Allow corrected land use to be applied to all scenarios

### PR DESCRIPTION
**Pull request by mmcfarland**
#### Overview
All modifications made or removed from the current conditions scenario are propagated across all other scenarios, including new and duplicated scenarios.  This facilitates using current conditions as a rectified base layer for constant model data. Both shared and scenario-specific modifications are submitted to the model endpoint. See detailed description in #87.

Additionally, the modification popups were altered to allow different behaviors for each type and the enhancement modifications were changed to types which have attributes more likely to affect the model output.

Connects #87 

screenshot from 2016-12-30 14 16 19

#### Known issues
There is a bug present on develop as well as this branch that cancels jobs too eagerly when switching between tabs if one is currently polling.  This can result in too many requests being generated for the number of scenarios there are.  Rather than try and fix it here, I'm going to investigate that in another issue, along with the aspects of the connected issue dealing with managing requests when modifications are shared.

#### Testing
* Update your local bundle `./scripts/bundle.sh --debug`
* Draw an area of interest and move to the modelling screen.
* To avoid the bug listed above, wait until polling completes before switching tabs, although all the intended behavior in this PR is not specifically affected - just the model results polling.
* Draw some land use changes on CC
    * Popup should be "normal", ie, let you delete
* Switch to new scenario, CC modification should be present
    * Has muted symbology
    * Popup won't let you delete, tells you where it's from
* Make new scenario, should behave like scenario above with CC changes already applied
* Duplicate a scenario, changes should be applied
* Delete a change on CC
    * Feature should be removed from all scenarios
* Scenario specific modifications should stay only available on that scenario, except when duplicated

### CODE REVIEW

**Comment by rajadain** [comment]
Since modifications drawn on Current Conditions will apply to all scenarios, consider changing this wording to say "all scenarios" instead of "this scenario"

image

**Comment by rajadain** [comment]
Added modification (blue, Buckwheat) doesn't show up in the scenario's baseline in the chart or table, while basemap data (gree, Orchard and red, Almonds) do:

image

The values of basemap data is also incorrect in the new scenario.

This functionality was just added in #93 by @arottersman. She may be able to help out here, or in a future separate card.

**Comment by rajadain** [comment]
If I add a new scenario after having made modifications in Current Conditions, it works correctly:

image

**Comment by rajadain** [comment]
I duplicated this scenario:

image

but while it had the right shapes, it didn't have the right results:

image

**Comment by rajadain** [comment]
Other tests worked well. It seems that the map / shape components are behaving as they should, but the results are not. Going to take a look at the code now.

**Comment by mmcfarland** [comment]
Yeah, I suspect the results being off are a result of the polling bug that I mentioned.  I think there may be significant rework to get that all aligned and my inclination is to not muddy this PR with those changes.  If the modifications are added/removed from scenarios as they should be, I'm willing to look at the polling/results issue separately.

File: src/icp/js/src/modeling/models.js [hunk 1]
```diff
@@ -127,6 +127,21 @@ var ProjectModel = Backbone.Model.extend({
...
+        this.get('scenarios').on('change:modification_hash', this.shareGlobalModifications);
+    },
+
+    shareGlobalModifications: function(changedScenario) {
+        // When modifications are made to current conditions, add them
+        // to all other scenarios as well
+        var scenarios = this;
+        if (changedScenario.get('is_current_conditions')) {
+            scenarios.each(function(scenario) {
+                var sharedMods = changedScenario.get('modifications').toJSON();
```
Comment (rajadain) [inline src/icp/js/src/modeling/models.js]: Consider initializing `sharedMods` outside the `.each` loop since it'll be the same for every occurrence

**Comment by rajadain** [comment]
The shapes are added / cloned / removed correctly, and the modeling runner can be fixed in a future PR.

**Comment by mmcfarland** [comment]
> Will there be another card that manages saving these shared modifications to the database?

That was actually an oversight.  I'm inclined to not save them specifically and instead keep the CC scenario be the only source of truth (where they are already saved), but I'll add a commit that re-initializes the field when the project is reloaded.  Good catch!

**Comment by mmcfarland** [comment]
>  consider changing this wording to say "all scenarios" instead of "this scenario"

The wording is correct when not on current conditions, but incorrect on it.  I'll add a conditional that changes it depending on the tab that is active.

**Comment by mmcfarland** [comment]
Added some new commits that address the feedback, specific new behavior:
* Popups on CC for drawn modifications read differently that those for drawn modifications on non CC scenarios.
* Reloading a saved project should result in shared modifications being distributed to scenarios again.

**Comment by arottersman** [comment]
Tested this pre-fixups, and it's looking good! I think I experienced the same shape/chart mismatch as @rajadain. In my case it was that I added some land cover to the current conditions after already having a few modifications. While the new mod was polling I switched to a scenario with its own existing modification. I could see the new current conditions shape, but it wasn't represented in the chart because the scenario chart didn't re-render once the current condition polling was complete. When I clicked the same scenario tab I was on again though, the chart refreshed and everything was correct.

**Comment by rajadain** [comment]
Saved a project and reloaded it, the shared modifications were applied correctly. Great work! Friendly reminder to squash before merging

**Pull request merged**